cpmpy/expressions/core.py

```diff
@@ -407,7 +407,7 @@ def __init__(self, name, arg_list):
         arity, is_bool = Operator.allowed[name]
         if is_bool:
```

Comment:
This is actually a hack (which I also applied in #291...) The arguments of an operator can be non-boolean while the output is. Not currently but maybe future operators will be.
We could make a new function on the `Expression` object "args_are_bool" or something like that. Or we make it a helper function in utils?

Reply:
If we don't have a concrete example, we don't support it nor future-proof for it; it would lead us too far

Reply:
if such an operator arrives, we just refactor to support it at that time